main.go

```diff
@@ -1,32 +1,39 @@
 package sdk
 
 import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"net"
 	"net/http"
 
 	"connectrpc.com/connect"
 	"github.com/utxorpc/go-codegen/utxorpc/v1alpha/query/queryconnect"
 	"github.com/utxorpc/go-codegen/utxorpc/v1alpha/submit/submitconnect"
 	"github.com/utxorpc/go-codegen/utxorpc/v1alpha/sync/syncconnect"
 	"github.com/utxorpc/go-codegen/utxorpc/v1alpha/watch/watchconnect"
+	"golang.org/x/net/http2"
 )
 
 type UtxorpcClient struct {
 	httpClient connect.HTTPClient
 	baseUrl    string
+	apiKey     string
 	ChainSync  syncconnect.ChainSyncServiceClient
 	Query      queryconnect.QueryServiceClient
 	Submit     submitconnect.SubmitServiceClient
 	Watch      watchconnect.WatchServiceClient
 }
 
-func NewClient(httpClient *http.Client, baseUrl string) UtxorpcClient {
+func NewClient(httpClient *http.Client, baseUrl string, apiKey string) UtxorpcClient {
 	var client UtxorpcClient
 	chainSyncClient := syncconnect.NewChainSyncServiceClient(httpClient, baseUrl, connect.WithGRPC())
 	queryClient := queryconnect.NewQueryServiceClient(httpClient, baseUrl, connect.WithGRPC())
 	submitClient := submitconnect.NewSubmitServiceClient(httpClient, baseUrl, connect.WithGRPC())
 	watchClient := watchconnect.NewWatchServiceClient(httpClient, baseUrl, connect.WithGRPC())
 	client.httpClient = httpClient
 	client.baseUrl = baseUrl
+	client.apiKey = apiKey
```

Comment:
This is currently implementation-specific (demeter) and isn't required by UTxO RPC. This should be a more generic "request headers" implementation if it's going into the main SDK code.